pkgs/os-specific/linux/nvidia-x11/generic.nix

```diff
@@ -23,35 +23,42 @@
 with stdenv.lib;
 
 assert (!libsOnly) -> kernel != null;
+assert (versionOlder version "391") -> sha256_32bit != null;
+assert !(versionOlder version "391") -> stdenv.hostPlatform.system == "x86_64-linux";
 
 let
   nameSuffix = optionalString (!libsOnly) "-${kernel.version}";
   pkgSuffix = optionalString (versionOlder version "304") "-pkg0";
+  i686bundled = (versionAtLeast version "391");
+
 
   self = stdenv.mkDerivation {
     name = "nvidia-x11-${version}${nameSuffix}";
 
     builder = ./builder.sh;
 
     src =
-      if stdenv.hostPlatform.system == "i686-linux" then
-        fetchurl {
-          url = "https://download.nvidia.com/XFree86/Linux-x86/${version}/NVIDIA-Linux-x86-${version}${pkgSuffix}.run";
-          sha256 = sha256_32bit;
-        }
-      else if stdenv.hostPlatform.system == "x86_64-linux" then
+      if stdenv.hostPlatform.system == "x86_64-linux" then
```

Comment:
Why was the order of i686 and x86_64 switched here?

Reply:
If I remember what happened correctly, my system seemed to always evaluate i686 as true for some reason so I never got up to x86_64.

Reply:
I think in some intermediate commits, there was an additional check, which made this order more logical. I have tested with the reverted order and it works as well.

Reply:
Alright, I'll try changing it back tomorrow if the PR isn't merged by then.